dht_options.go

```diff
@@ -309,3 +309,24 @@ func forceAddressUpdateProcessing(t *testing.T) Option {
 		return nil
 	}
 }
+
+// EnableOptimisticProvide enables and optimization that skips the last hops of the provide process.
+// This works by using the network size estimator (which use the keyspace density of queries)
+// to optimistically send ADD_PROVIDER requests when we most likely find the last hops.
+// It will also run some ADD_PROVIDER requests asynchronously in the background after returning,
+// this allows to optimistically return earlier if some threshold succeeded.
```

Comment:
can you make it clear here that this is experimental and could be removed?

Reply:
done 👍